cip/1.accepted/CIP2017-04-13-Aggregations.adoc

```diff
@@ -0,0 +1,110 @@
+= CIP2017-04-13 Aggregations
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Tobias Lindaaker <tobias.lindaaker@neotechnology.com>
+
+toc::[]
+
+== Aggregations
+
+=== Syntax
+
+[source, ebnf]
+----
+Return = 'RETURN', ['DISTINCT'], ReturnBody, Filter ;
+With = 'WITH', ['DISTINCT'], ReturnBody, Filter ;
+SingleValueReturn = 'RETURN', (Expression | ProjectedMap | Aggregation), Filter ;
+
+ReturnBody = ('*' | ReturnItem), {',', ReturnItem} ;
+ReturnItem = (Expression, ['AS', Variable])
+           | (Aggregation, ['AS', Variable])
+           | (ProjectedMap, 'AS', Variable)
+           ;
+
+Filter = [Where], [Order], [Skip], [Limit] ;
+
+Aggregation = Aggregator, 'OF', Expression ;
+Aggregator = SymbolicName | ExtensionName ;
+ExtensionName = {SymbolicName, '.'}-, SymbolicName ;
+----
+
+=== Examples
+
+[source, cypher]
+.Aggregation using `avg`
+----
+MATCH (employee:Employee)
+RETURN avg OF employee.salary
+----
+
+[source, cypher]
+.Aggregation using `collect`
+----
+MATCH (person:Person)-[:FRIEND]-(friend)
+RETURN person.email, collect OF friend {.name,.email} AS friends
+----
+
+[source, cypher]
+.Aggregation using `count`
+----
+MATCH (nodes) RETURN count OF nodes
+----
+
+[source, cypher]
+.Aggregation using `max`
+----
+MATCH (person:Person)-[:LIVES_IN]->({country:$country})
+RETURN max OF person.age
+----
+
+[source, cypher]
+.Aggregation using `min`
+----
+MATCH (movie:Movie)
+RETURN min OF movie.duration
+----
+
+[source, cypher]
+.Aggregation using `percentileCont`
+----
+BREAKS DOWN IN THIS SYNTAX
```

Comment:
Nice ^^ 

How about - so we have a few options to give food for thought - a parentheses-free version:

`
MATCH (n)
RETURN percentileCont OF n.property GIVEN $percentile
`
I also played around with using `WITH` instead of `GIVEN`, but that is *way* too overloaded.

If we ever have aggregations with more than one argument of this type, e.g. `myAgg(expression, arg1, arg2)` in today's syntax, these are the options we'd have (continuing with the same examples as above):

1. `... RETURN myAgg($arg1, @arg2) OF n.property`
2. `... RETURN myAgg OF n.property GIVEN $arg1, @arg2`